### ISSUE TITLE
linter: `jsdoc/require-param`

### ISSUE THREAD

**Issue by JakobJingleheimer** [oxc-project/oxc#10253]
### What version of Oxlint are you using?

0.16.4

### What command did you run?

`oxlint --config ./.oxlintrc.json5`

### What does your `.oxlintrc.json` config file look like?

<details>
<summary><code>.oxlintrc.json5</code></summary>

```json5
{
	"ignorePatterns": [
		"**/fixtures/**",
		"**/fixture.*",
		"**/*.snap.cjs",
	],
	"categories": {
		"correctness": "error",
		"perf": "error",
		"pedantic": "error",
		"suspicious": "warn",
	},
	"plugins": [
		"import",
		"jsdoc",
		"jsx-a11y",
		"node",
		"oxc",
		"react-perf",
		"react",
		"recommended",
		"typescript",
		"unicorn",
	],
	"rules": {
		"eslint/no-case-declarations": "off",
		"eslint/no-constructor-return": "off",
		"eslint/require-await": "off",
		"import/max-dependencies": "off",
		"eslint/max-depth": ["error", 3],
		"eslint/max-lines-per-function": ["error", 30],
		"eslint/max-nested-callbacks": ["error", 3],
		"jsdoc/require-returns": "off",
		"jsdoc/require-returns-description": "off",
		"no-console": "error",
		"no-import-type-side-effects": "error",
		"typescript/ban-types": "off",
		"unicorn/explicit-length-check": "off",
		"unicorn/no-typeof-undefined": "off",
	},
	"overrides": [
		{
			"files": [
				"*.spec.mjs",
				"*.spec.jsx",
				"*.test.mjs",
				"*.test.jsx",
			],
			"rules": {
				"max-depth": "off",
				"max-lines-per-function": "off",
				"max-nested-callbacks": "off",
			},
		},
	],
}

```
</details>

### What happened?

Functions annotated with `@type` do not need any other jsdocs, yet related rules still error:

```js
  × eslint-plugin-jsdoc(require-param): Missing JSDoc `@param` declaration for function parameters.
    ╭─[packages/jsonc/jsonc.loader.mjs:9:29]
  7 │  * @type {import('node:module').ResolveHook}
  8 │  */
  9 │ async function resolveJSONC(specifier, ctx, nextResolve) {
    ·                             ─────────  ───  ───────────
 10 │     const nextResult = await nextResolve(specifier);
    ╰────
  help: Add `@param` tag with name.
```
